apollo-router/src/plugins/telemetry/config_new/http_client/selectors.rs

```diff
@@ -0,0 +1,178 @@
+use derivative::Derivative;
+use schemars::JsonSchema;
+use serde::Deserialize;
+use tower::BoxError;
+
+use crate::Context;
+use crate::plugins::telemetry::config_new::Selector;
+use crate::plugins::telemetry::config_new::Stage;
+use crate::plugins::telemetry::config_new::instruments::InstrumentValue;
+use crate::plugins::telemetry::config_new::instruments::Standard;
+use crate::services::http;
+
+#[derive(Deserialize, JsonSchema, Clone, Debug)]
+#[serde(deny_unknown_fields, rename_all = "snake_case", untagged)]
+pub(crate) enum HttpClientValue {
+    Standard(Standard),
+    Custom(HttpClientSelector),
+}
+
+impl From<&HttpClientValue> for InstrumentValue<HttpClientSelector> {
+    fn from(value: &HttpClientValue) -> Self {
+        match value {
+            HttpClientValue::Standard(standard) => InstrumentValue::Standard(standard.clone()),
+            HttpClientValue::Custom(selector) => InstrumentValue::Custom(selector.clone()),
+        }
+    }
+}
+
+#[derive(Derivative, Deserialize, JsonSchema, Clone)]
+#[serde(deny_unknown_fields, untagged)]
+#[derivative(Debug, PartialEq)]
+pub(crate) enum HttpClientSelector {
+    /// A header from the HTTP request
+    HttpClientRequestHeader {
+        /// The name of the HTTP client request header.
+        http_client_request_header: String,
```

Comment:
```suggestion
       request_header: String,
```
Let's just name this request header. We're not exposing the originating request info.